js/components/thumbnail.js

```diff
@@ -7,12 +7,107 @@ var React = require('react'),
     Utils = require('./utils');
 
 var Thumbnail = React.createClass({
+  getInitialState: function() {
+    this.positionY = 0;
+    this.positionX = -320;
+    this.imageWidth = 0;
```

Comment:
What is the difference between imageWidth and thumbnailWidth?

Reply:
thumbnailWidth is width of thumbnail,
imageWidth is width of image for thumbnail, so if this image is image for 360, i.e. sphire, the width can be bigger than thumbnailWidth